static/app/views/issueDetails/streamline/issueCronCheckTimeline.tsx

```diff
@@ -0,0 +1,243 @@
+   * are not available (e.g. time range has changed and page refreshed)
+   */
+  const user = useUser();
+  const organization = useOrganization();
+  const {detectorDetails} = useIssueDetails();
+  const {detectorId, detectorType} = detectorDetails;
+
+  const hasCronDetector = detectorId && detectorType === 'cron_monitor';
+
+  const {data: event} = useApiQuery<Event>(
+    getGroupEventQueryKey({
+      orgSlug: organization.slug,
+      groupId,
+      eventId: user.options.defaultIssueEvent,
+      environments: [],
+    }),
+    {
+      staleTime: Infinity,
+      enabled: !hasCronDetector,
+      retry: false,
+    }
+  );
+
+  // Fall back to the fetched event since the legacy UI isn't nested within the provider the provider
+  return hasCronDetector
+    ? detectorId
+    : event?.tags?.find(({key}) => key === 'monitor.id')?.value;
+}
+
+function useCronLegendStatuses({bucketStats}: {bucketStats: MonitorBucket[]}) {
+  /**
+   * Extract a list of statuses that have occurred at least once in the bucket stats.
+   */
+  return useMemo(() => {
+    const statusMap: Record<CheckInStatus, boolean> = {
+      [CheckInStatus.OK]: true,
+      [CheckInStatus.ERROR]: false,
+      [CheckInStatus.IN_PROGRESS]: false,
+      [CheckInStatus.MISSED]: false,
+      [CheckInStatus.TIMEOUT]: false,
+      [CheckInStatus.UNKNOWN]: false,
+    };
+    bucketStats?.forEach(([_timestamp, bucketEnvMapping]) => {
+      const bucketEnvMappingEntries = Object.values(bucketEnvMapping) as Array<
+        StatsBucket<CheckInStatus>
+      >;
+      for (const statBucket of bucketEnvMappingEntries) {
+        const statBucketEntries = Object.entries(statBucket) as Array<
+          [CheckInStatus, number]
+        >;
+        for (const [status, count] of statBucketEntries) {
+          if (count > 0 && !statusMap[status]) {
+            statusMap[status] = true;
+          }
+        }
+      }
+    });
+    return (Object.keys(statusMap) as CheckInStatus[]).filter(
+      status => statusMap[status]
+    );
+  }, [bucketStats]);
+}
+
+export function IssueCronCheckTimeline({group}: {group: Group; event?: Event}) {
+  const elementRef = useRef<HTMLDivElement>(null);
+  const {width: containerWidth} = useDimensions<HTMLDivElement>({elementRef});
+  const timelineWidth = useDebouncedValue(containerWidth, 500);
+  const timeWindowConfig = useTimeWindowConfig({timelineWidth});
+
+  const cronAlertId = useCronIssueAlertId({groupId: group.id});
+  const {data: stats, isPending} = useMonitorStats({
+    monitors: cronAlertId ? [cronAlertId] : [],
+    timeWindowConfig,
+  });
+
+  const cronStats = useMemo(() => {
+    if (!cronAlertId) {
+      return [];
+    }
+    return stats?.[cronAlertId] ?? [];
+  }, [cronAlertId, stats]);
+
+  const statEnvironments = useMemo(() => {
+    const envSet = cronStats.reduce((acc, [_, envs]) => {
+      return new Set([...acc, ...Object.keys(envs)]);
+    }, new Set<string>());
```

Comment:
would mutate the accumulator instead of making a new one every time in reduce
```suggestion
    const envSet = cronStats.reduce((acc, [_, envs]) => {
      Object.keys(envs).forEach(env => acc.add(env));
      return acc;
    }, new Set<string>());
```

Reply:
makes sense, i was honestly surprised there was no built in method for adding a list to a set

Reply:
yeah i thought add took multiple values but here we are